### PR TITLE
Adding backpropagation in deep learning

### LINKED ISSUE #3129: [Feature Request]: Adding backpropogation in Deep Learning
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

This feature will ensure hands-on approach helps users understand the mechanics of backpropagation, including gradient descent, weight updates, and error calculation, offering a deeper insight into how neural networks learn.

### Use Case

Ideal for students, educators, and developers, this feature provides a clear understanding of neural network training. Students can explore the step-by-step process, educators can demonstrate the algorithm in classes, and developers can build custom solutions, enhancing their knowledge and flexibility in neural network design.

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Shantnu-singh**
## Description

Added docs on backpropagation in deep learning. solving is issue numbered 

Fix #3129.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [ ] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [x] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [x] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [x] [Markdown Guide](https://www.markdownguide.org/)
- [x] [MDX Documentation](https://mdxjs.com/docs/)
- [x] [Mermaid Documentation](https://mermaid.js.org/)

### CODE REVIEW

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
@Shantnu-singh, I'm afraid there's an issue with your pull request. The images are way too big, and we need to optimize the solution. You can delete the images and try using our feature https://codeharborhub.github.io/community/features/#math-equations-with-katex for the math equations.

**Comment by Shantnu-singh** [comment]
hey @Ajay-Dhangar can you review this and let me know of any further changes.

**Review by Ajay-Dhangar — APPROVED** [review]
Done
